### PR TITLE
Fixes #46764 API Documentation

### LINKED ISSUE #46764: Rest api documentation #documentation
https://guides.rubyonrails.org/api_app.html#what-is-an-api-application-questionmark

I read the API documentation but seem very confused, I hope I can see the documentation about creating simple restapi and authenticator in the doc.

### PR DESCRIPTION
**Pull request by davidomullan**
### Motivation / Background

Fixes #46764 by adding further explanation of controllers in API only mode (via example).

### Detail

Guide for API Rails app is lacking example of what controllers look like in API-only mode. Added example of generating resource in API-only and viewing JSON output.

### Checklist

Before submitting the PR make sure the following are checked:

* [ x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [n/a ] Tests are added or updated if you fix a bug or add a feature.
* [n/a ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/api_app.md [hunk 1]
```diff
@@ -151,6 +151,113 @@ This will do three main things for you:
...
+# app/models/member.rb
+class Member < ApplicationRecord
+  belongs_to :group
+end
+```
+
+Now if we open our Groups Controller, we should notice that with an API rails
+app we are rendering JSON data only. On the index action we query for Group.all
+and assign it an instance variable called @groups. That will automatically return
+a JSON object containing groups.
```
Comment (p8) [inline guides/source/api_app.md:187]: @groups doesn't return a JSON object. It gets turned into JSON by the render call.
Maybe something like:
```suggestion
and assign it an instance variable called `@groups`. Passing it to `render` with the
`:json` option will automatically render the groups as JSON.
```

**Review by p8 — CHANGES_REQUESTED** [review]
I don't have an opinion about wether this should be added to the guides, but I made some small suggestion on the form.

**Comment by skipkayhil** [comment]
There's nothing to do about the "Changes Requested", when Petrik has time to review again his approval will replace it.

In the meantime could you [squash your commits](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits)?

**Comment by davidomullan** [comment]
> There's nothing to do about the "Changes Requested", when Petrik has time to review again his approval will replace it.
> 
> In the meantime could you [squash your commits](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits)?

Thank you very much for the help, @skipkayhil ! I was really racking my brain on how to squash commits and then realized you linked the answer. Rebasing to the upstream/main fixed everything.

File: guides/source/api_app.md [hunk 2]
```diff
@@ -151,6 +151,113 @@ This will do three main things for you:
...
+# app/models/member.rb
+class Member < ApplicationRecord
+  belongs_to :group
+end
+```
+
+Now if we open our Groups Controller, we should notice that with an API rails
+app we are rendering JSON data only. On the index action we query for Group.all
+and assign it an instance variable called `@groups`. Passing it to `render` with the
+`:json` option will automatically render the groups as JSON.
```
Comment (p8) [inline guides/source/api_app.md:187]: Sorry, still some tiny comments:
```suggestion
Now if we open our `GroupsController`, we should notice that with an API Rails
app we are rendering JSON data only. On the index action we query for `Group.all`
and assign it to an instance variable called `@groups`. Passing it to `render` with the
`:json` option will automatically render the groups as JSON.
```

File: guides/source/api_app.md [hunk 3]
```diff
@@ -151,6 +151,113 @@ This will do three main things for you:
...
+```
+
+Finally we can add some groups to our database.
+
+```bash
+bin/rails console
+
+Group.create(name: "Rails Founders")
+Group.create(name: "Rails Contributors")
+```
```
Comment (p8) [inline guides/source/api_app.md:251]: I'm guessing if someone managed to get this far, they know to start the console.
I think we can use the `irb` codefences similar to the getting [started guide](https://edgeguides.rubyonrails.org/getting_started.html#using-a-model-to-interact-with-the-database)
```suggestion
Finally we can add some groups to our database from the Rails console:

```irb
irb>Group.create(name: "Rails Founders")
irb> Group.create(name: "Rails Contributors")
```
```

Or maybe it's clearer if we still mention `bin/rails`:
```suggestion
Finally we can add some groups to our database by running `bin/rails console`:

```irb
irb>Group.create(name: "Rails Founders")
irb> Group.create(name: "Rails Contributors")
```
```

File: guides/source/api_app.md [hunk 4]
```diff
@@ -151,6 +151,113 @@ This will do three main things for you:
...
+
+With some data in the app, we can boot up the server and see our JSON data.
+```bash
+bin/rails server
+```
```
Comment (p8) [inline guides/source/api_app.md:256]:
```suggestion
With some data in the app, we can boot up the server and visit <http://localhost:3000/groups.json> to see our JSON data.

```

File: guides/source/api_app.md [hunk 5]
```diff
@@ -151,6 +151,113 @@ This will do three main things for you:
...
+```json
+[{"id":1,"name":"Rails Founders","created_at":"yyyy-MM-ddTHH:mm:ss.SSZ","updated_at":"yyyy-MM-ddTHH:mm:ss.SSZ"},{"id":2,"name":"Rails Contributors","created_at":"yyyy-MM-ddTHH:mm:ss.SSZ","updated_at":"yyyy-MM-ddTHH:mm:ss.SSZ"}]
```
Comment (p8) [inline guides/source/api_app.md:258]: I think it's more readable if we add some linkbreaks (even if those aren't present in the actual JSON).
```suggestion
[
{"id":1, "name":"Rails Founders", "created_at":" ...},
{"id":2, "name":"Rails Contributors", "created_at": .... }
]
```

**Review by p8 — CHANGES_REQUESTED** [review]

File: guides/source/api_app.md (hunk 3 above)
Comment (p8) [reply guides/source/api_app.md:251]: Or maybe it's clearer if we still mention `bin/rails`:

```suggestion
Finally we can add some groups to our database by running `bin/rails console`:

```irb
irb>Group.create(name: "Rails Founders")
irb> Group.create(name: "Rails Contributors")
```
```

File: guides/source/api_app.md (hunk 2 above)
Comment (davidomullan) [reply guides/source/api_app.md:187]: No apologies necessary, thank you for the help! I will try to keep these formatting changes in mind for any future contributions. Still a lot to learn :D

**Comment by p8** [comment]
Thanks @davidomullan !
Giving this a final read, I think we should remove creating the member resources as it doesn't adds much to the example.

**Comment by davidomullan** [comment]
@p8 agreed; I had originally intended to provide a more complete example but it seemed overkill. I just never ended up removing members.

Will fix later today and recommit.

**Comment by p8** [comment]
Thanks @davidomullan !

**Pull request merged**